### DISCUSSION TITLE
After updating to 2.1.0 i am facing a weird issue while scolling it is showing wrong image.Help Required mean while it is working fine with 1.4.0

### DISCUSSION THREAD

**Discussion by zeeshanrasool91** [coil-kt/coil#1286] [General]
**Describe the bug**
A clear and concise description of what the bug is. **Please [enable logging](https://coil-kt.github.io/coil/faq/#how-do-i-enable-logging) and attach any stack traces.** Bug reports without reproduction steps are likely to be closed.

**To Reproduce**
How can we reproduce this? Please attach a small project that reproduces the bug.

**Logs/Screenshots**
If applicable, [add logs](https://coil-kt.github.io/coil/faq/#how-do-i-enable-logging) or screenshots to help explain your problem.

**Version**
What library version are you using? Does this occur on a specific API level or Android device?

Please refer to this video for detail.

https://drive.google.com/file/d/1mEqdh9qse72Bd7sOVaDKZSVIJsuhpfiu/view?usp=sharing

i am using simple extension method to load image imageview.load()

**Comment by zeeshanrasool91** [comment]
Using this library to show sticky headers in recycler view it was working fine till 1.4.0 but after updating it to 2.1.0 it is giving this weird behaviour. please guide how i can provide you with more details.i have simple recycler view adapter, populating data like this
in bindview holder

val section = items[position] as SectionItem
                val sectionHolder = viewHolder as SectionViewHolderViaBinding
                sectionHolder.binding.imgFlag.loadImage(section.sectionImage)
                sectionHolder.binding.textTitle.text = section.sectionTitle
this.load(drawable)

**Reply by zeeshanrasool91** [reply]
https://drive.google.com/file/d/19WWJd_dfKN7YJPtPEoCpLhRAO3575KxW/view?usp=sharing 
Please refer this URL for project, if i will downgrade project to 1.4.0 it will work fine

**Reply by colinrtwhite** [reply]
Thanks for the repro project. I'll take a look.

**Comment by colinrtwhite** [comment]
@zeeshanrasool91 The request for the item header is being cancelled automatically as the view isn't attached to a window. Due to the way you're drawing the header view you'll either need to attach it to a parent view first or update the section image request to:

```kotlin
val context = sectionHolder.binding.imgFlag.context
val request = ImageRequest.Builder(context)
    .data(section.sectionImage)
    .target(onSuccess = sectionHolder.binding.imgFlag::setImageDrawable)
    .build()
context.imageLoader.enqueue(request)
```

**Comment by zeeshanrasool91** [comment]
Sure thanks i will try

Regards
Zeeshan Rasool
________________________________

**Reply by zeeshanrasool91** [reply]
The mentioned code solve the issue, a little more help will be appreciated if you can explain me little bit about how i can attach it to parent view. because i am using one of the gist to achieve this header effect.

i am using this library

https://github.com/filipkowicz/HeaderItemDecorationExample

Here is the gist file which is used by this project.

https://gist.github.com/filipkowicz/1a769001fae407b8813ab4387c42fcbd
